webserver.py

```diff
@@ -35,84 +35,82 @@ def get_conn():
-
     for i in range(GAME_CONFIG['field_width']):
         for j in range(GAME_CONFIG['field_height']):
-            if game_state.get("plot" + str(i) + "," + str(j)) is None:
-                game_state["plot" + str(i) + "," + str(j)] = {'seedType': 0, 'sowTime': 0, 'locked': 1}
+            if game_state.get('plot' + str(i) + '_' + str(j)) is None:
+                game_state['plot' + str(i) + '_' + str(j)] = {'seedType': 0, 'sowTime': 0, 'locked': 1}
+            if get_plot(game_state, i, j)['seedType'] != 0:
+                total_crops_planted += 1
     for i in range(2):
         for j in range(2):
-            game_state[("plot" + str(i) + "," + str(j))]['locked'] = 0
-
-    total_seed_count = 0
-    total_crops_planted = 0
-    for seed in GAME_CONFIG['seeds']:
-            total_seed_count += game_state['seedCounts'][seed]
-    for i in range(GAME_CONFIG['field_width']):
-        for j in range(GAME_CONFIG['field_height']):
-            if game_state["plot" + str(i) + "," + str(j)]['seedType'] != 0:
-               total_crops_planted += 1
-    if total_seed_count == 0 and total_crops_planted == 0 and game_state['resources']['cash'] < GAME_CONFIG['starting_resources'][resource]:
+            get_plot(game_state, i, j)['locked'] = 0
+    if total_seed_count == 0 and total_crops_planted == 0 \
+            and game_state['resources']['cash'] < GAME_CONFIG['starting_resources'][resource]:
         game_state['resources']['cash'] = GAME_CONFIG['starting_resources']['cash']
     return game_state
 
 
 def save_state(slug, game_state):
     my_conn = get_conn()
     json_game_state = json.dumps(game_state, indent=4)
-    with my_conn:
-        with my_conn.cursor() as cursor:
-            cursor.execute("INSERT INTO game_states (slug, game_state) VALUES (%s, %s) ON CONFLICT (slug) DO UPDATE SET game_state=%s;",
-                           [slug, json_game_state, json_game_state])
+    with my_conn.cursor() as cursor:
+        cursor.execute('INSERT INTO game_states (slug, game_state) VALUES (%s, %s) '
+                       'ON CONFLICT (slug) DO UPDATE SET game_state=%s;',
+                       [slug, json_game_state, json_game_state])
 
             
 def get_leaderboard_data():
     my_conn = get_conn()
-    with my_conn:
-        with my_conn.cursor() as cursor:
-            cursor.execute("SELECT slug, game_state->'resources'->'cash' AS cash, game_state->'seedCounts'->'m' AS m FROM game_states ORDER BY 2 DESC LIMIT 10;")
-            result = cursor.fetchall()
-            return result
+    with my_conn.cursor() as cursor:
+        cursor.execute("SELECT slug, game_state->'resources'->'cash' AS cash, "
+                       "game_state->'seedCounts'->'m' AS m FROM game_states ORDER BY 2 DESC LIMIT 10;")
+        result = cursor.fetchall()
+        return result
 
 
 def get_admin_data():
     data = {}
     my_conn = get_conn()
-    with my_conn:
```

Comment:
The `with my_conn:` is necessary.  See:
http://initd.org/psycopg/docs/usage.html#with-statement
and the red box under
http://initd.org/psycopg/docs/usage.html#transactions-control